server/views.py

```diff
@@ -182,6 +183,95 @@ def card(request, tool_id, card_id):
 
   return makeResponse(request, result)
 
+@check_secret
+@check_ip
+@require_GET
+def getinfo(request, tool_id, item_requested):
+  ip = get_ip(request)
+  try:
+    t = Tool.objects.get(pk=tool_id)
+  except ObjectDoesNotExist as e:
+    logger.warning('Tool does not exist for %s // %s from %s', request.method, request.path, ip,
+                extra={
+                    'status_code': 200,
+                    'request': request
+                }
+            )
+    result = { 'numeric_status' : -1, 'error' : 'Tool does Not Exist' }
+    return makeResponse(request, result)
+
+  try:
+    i = Venditem.objects.get(item=item_requested)
+  except ObjectDoesNotExist as e:
+    logger.warning('Item does not exist %s // %s from %s', request.method, request.path, ip,
+                extra={
+                    'status_code': 200,
+                    'request': request
+                }
+            )
+    result = { 'numeric_status' : -1, 'error' : 'Item does Not Exist' }
+    return makeResponse(request, result)
+
+  result = { 'numeric_status' : 1 }
+  # For subscribed users, getting the name is useful for doorbot
+  result['item_name'] = i.name
+  result['item'] = i.item
+  result['price'] = i.price
+  result['stock'] = i.stock
+
+  return makeResponse(request, result)
+
+
+@check_secret
+@check_ip
+@require_GET
+def updatebalance(request, tool_id, new_balance, card_id):
+  try:
+    t = Tool.objects.get(pk=tool_id)
+  except ObjectDoesNotExist as e:
+    result = { 'numeric_status' : -1, 'error' : 'Tool does not exist'}
+    return makeResponse(request, result)
+
+  try:
+    c = Card.objects.get(card_id=card_id)
+  except ObjectDoesNotExist as e:
+    result = { 'numeric_status' : 0, 'error' : 'Card does not exist'}
+    return makeResponse(request, result)
+
+  if not c.user.subscribed:
+    # needs to be subscribed to be a user
+    result = { 'numeric_status' : 0, 'error' : 'Card is not subscribed'}
+    return makeResponse(request, result)
+
+  c.user.balance = new_balance
+  c.user.save()
+
+
+  result = { 'numeric_status' : 1, 'success' : 'Balance Updated'}
+  l = Log(tool=t, user=c.user, message="Balance updated to : " + new_balance)
+  l.save()
+  return makeResponse(request, result)
+
+@check_secret
+@check_ip
+@require_GET
+def updatestock(request, tool_id, item_requested, new_stock):
+  try:
+    t = Tool.objects.get(pk=tool_id)
+  except ObjectDoesNotExist as e:
+    result = { 'numeric_status' : -1, 'error' : 'Tool does not exist'}
+    return makeResponse(request, result)
+  try:
+    i = Venditem.objects.get(item = item_requested)
+  except ObjectDoesNotExist as e:
+    result = { 'numeric_status' : -1, 'error' : 'Item does not exist'}
+    return makeResponse(request, result)
+
+  i.stock = int(new_stock)
```

Comment:
I feel like a log message would be handy here. In general logging in here kinda sucks so I want to rethink it a bit anyway.

Also, I wonder if this should be broken into separate "restock" (add only) and "stock count" (forces a new value into the stock field) since under normal circumstances I feel like you would only want to add, and if things aren't in sync, something has gone wrong so that should be a separate action to be explicit